lib/AST/ProtocolConformance.cpp

```diff
@@ -966,8 +966,23 @@ Witness SelfProtocolConformance::getWitness(ValueDecl *requirement,
 ConcreteDeclRef
 RootProtocolConformance::getWitnessDeclRef(ValueDecl *requirement,
                                            LazyResolver *resolver) const {
-  if (auto witness = getWitness(requirement, resolver))
-    return witness.getDeclRef();
+  if (auto witness = getWitness(requirement, resolver)) {
+    auto *witnessDecl = witness.getDecl();
+
+    // If the witness is generic, you have to call getWitness() and build
+    // your own substitutions in terms of the synthetic environment.
+    if (auto *witnessDC = dyn_cast<DeclContext>(witnessDecl))
```

Comment:
Would this be triggered by just another level of genericness in the crasher example?

Reply:
I’ll take a look. It would not have worked before anyway though.